src/coreclr/vm/stubgen.cpp

```diff
@@ -2763,6 +2763,69 @@ void ILStubLinker::SetStubTargetCallingConv(CorCallingConvention uNativeCallingC
     }
 }
 
+void ILStubLinker::SetStubTargetCallingConv(CorInfoCallConvExtension callConv)
+{
+    LIMITED_METHOD_CONTRACT;
+    CorCallingConvention originalCallingConvention = m_nativeFnSigBuilder.GetCallingConv();
+    if (originalCallingConvention != IMAGE_CEE_CS_CALLCONV_UNMANAGED)
+    {
+        // For performance reasons, we try to encode the calling convention using
+        // the flags-based method if possible before using modopts.
+        switch (callConv)
+        {
+            case CorInfoCallConvExtension::C:
+                m_nativeFnSigBuilder.SetCallingConv(IMAGE_CEE_CS_CALLCONV_C);
+                break;
+            case CorInfoCallConvExtension::Stdcall:
+                m_nativeFnSigBuilder.SetCallingConv(IMAGE_CEE_CS_CALLCONV_STDCALL);
+                break;
+            case CorInfoCallConvExtension::Thiscall:
+                m_nativeFnSigBuilder.SetCallingConv(IMAGE_CEE_CS_CALLCONV_THISCALL);
+                break;
+            case CorInfoCallConvExtension::Fastcall:
+                m_nativeFnSigBuilder.SetCallingConv(IMAGE_CEE_CS_CALLCONV_FASTCALL);
+                break;
+            default:
+                m_nativeFnSigBuilder.SetCallingConv(IMAGE_CEE_CS_CALLCONV_UNMANAGED);
+                break;
+        }
+    }
+
+    if (m_nativeFnSigBuilder.GetCallingConv() == IMAGE_CEE_CS_CALLCONV_UNMANAGED)
```

Comment:
I am leery of this side-effect setting logic. At a minimum add some comments to draw attention to the setting of his in the previous branch. The fact that line 2769 gets this and this subsequent gets a perhaps different value is less than clear.